crates/lib/src/qpu/experimental/randomized_measurements.rs

```diff
@@ -213,11 +213,14 @@ impl RandomizedMeasurements {
             .qubit_randomizations
             .iter()
             .filter(|randomization| measured_qubits.contains(&randomization.measurement.qubit))
-            .map(|randomization| randomization.measurement.qubit.clone())
+            .map(|randomization| &randomization.measurement.qubit)
             .collect::<HashSet<_>>();
         if !qubits_with_redundant_measurements.is_empty() {
             return Err(Error::ProgramContainsPreexistingMeasurements(
-                qubits_with_redundant_measurements,
+                qubits_with_redundant_measurements
+                    .into_iter()
+                    .cloned()
+                    .collect(),
```

Comment:
Oh, I missed that this was here 🤦🏻. Cloning in the initial iterator (line 216) makes sense now.